modules/lang-painless/src/main/java/org/elasticsearch/painless/DecorateExecutePhase.java

```diff
@@ -0,0 +1,310 @@
+            catchBlockNode.setStatementCount(1);
+
+            catchNode.setBlockNode(catchBlockNode);
+
+            ThrowNode throwNode = new ThrowNode();
+            throwNode.setLocation(internalLocation);
+
+            catchBlockNode.addStatementNode(throwNode);
+
+            UnboundCallNode unboundCallNode = new UnboundCallNode();
+            unboundCallNode.setLocation(internalLocation);
+            unboundCallNode.setExpressionType(ScriptException.class);
+            unboundCallNode.setLocalFunction(new LocalFunction(
+                            "convertToScriptException",
+                            ScriptException.class,
+                            Arrays.asList(Throwable.class, Map.class),
+                            true,
+                            false
+                    )
+            );
+
+            throwNode.setExpressionNode(unboundCallNode);
+
+            VariableNode variableNode = new VariableNode();
+            variableNode.setLocation(internalLocation);
+            variableNode.setExpressionType(ScriptException.class);
+            variableNode.setName("#painlessExplainError");
+
+            unboundCallNode.addArgumentNode(variableNode);
+
+            CallNode callNode = new CallNode();
+            callNode.setLocation(internalLocation);
+            callNode.setExpressionType(Map.class);
+
+            unboundCallNode.addArgumentNode(callNode);
+
+            variableNode = new VariableNode();
+            variableNode.setLocation(internalLocation);
+            variableNode.setExpressionType(PainlessExplainError.class);
+            variableNode.setName("#painlessExplainError");
+
+            callNode.setLeftNode(variableNode);
+
+            CallSubNode callSubNode = new CallSubNode();
+            callSubNode.setLocation(internalLocation);
+            callSubNode.setExpressionType(Map.class);
+            callSubNode.setBox(PainlessExplainError.class);
+            callSubNode.setMethod(new PainlessMethod(
+                    PainlessExplainError.class.getMethod(
+                            "getHeaders",
+                            PainlessLookup.class),
+                    PainlessExplainError.class,
+                    null,
+                    Collections.emptyList(),
+                    null,
+                    null,
+                    null
+            ));
+
+            callNode.setRightNode(callSubNode);
+
+            UnboundFieldNode unboundFieldNode = new UnboundFieldNode();
+            unboundFieldNode.setLocation(internalLocation);
+            unboundFieldNode.setExpressionType(PainlessLookup.class);
+            unboundFieldNode.setName("$DEFINITION");
+            unboundFieldNode.setStatic(true);
+
+            callSubNode.addArgumentNode(unboundFieldNode);
+
+            for (Class<?> throwable : new Class<?>[] {
+                    PainlessError.class, BootstrapMethodError.class, OutOfMemoryError.class, StackOverflowError.class, Exception.class}) {
```

Comment:
this could be a class constant, like `ERRORS_TO_CATCH`?

Reply:
This is going to change in future PRs so I would prefer to leave this for now.